landlab/components/landslides/landslide_probability.py

```diff
@@ -0,0 +1,535 @@
+#!/usr/env/python
+
+"""
+landslide_probability.py:
+
+Landlab component that simulates relative wetness and probability of failure.
+
+Relative wetness and factor-of-safety are based on the infinite slope
+stability model driven by topographic and soils inputs and recharge provided
+by user in a "landslide_driver" file. In addition, the component simulates
+the probability of failure for each node based on Monte Carlo simulations
+of the factor-of-safety as the number of simulations with factor-of-safety
+<= 1.0 divided by the number of simulations.
+
+Modified to be more generic in the inputs for greater usability as well
+as accomodate functionality with new release of Landlab version 1.
+
+.. codauthor:: R.Strauch, E.Istanbulluoglu, & S.Nudurupati
+University of Washington
+Created on Thu Aug 20, 2015
+Last edit April 04, 2017
+"""
+
+# %% Import Libraries
+from landlab import Component
+from landlab.utils.decorators import use_file_name_or_kwds
+import numpy as np
+from scipy import interpolate
+from statsmodels.distributions.empirical_distribution import ECDF
+import copy
+
+# %% Instantiate Object
+
+
+class LandslideProbability(Component):
+    """
+    Landlab component designed to calculate a probability of failure at
+    each grid node based on the infinite slope stability model
+    stability index (Factor of Safety).
+
+    The driving force for failure is provided by the user in the form of
+    groundwater recharge; 4 options for providing recharge are included.
+    The model uses topographic and soils characteristics provided as input
+    by the user in the landslide_driver.
+
+    A LandslideProbability calcuation function provides the user with the
+    mean soil relative wetness and probabilty of failure at each node.
+
```

Comment:
By  a LandslideProbability calculation function, do you mean to say something like:

The main method of the LandslideProbability class is calculate_landslide_probability() which calculates the mean soil relative wetness and probabilty of failure at each node based on a Monte Carlo approach.